part1/unicafe/src/App.jsx

```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react'
+
+// a proper place to define a component
+const Statistics = ({presentStatistics, good, neutral, bad}) => {
+  // ...
+
+  const total = {good} + {neutral} + {bad}
+
+  if (presentStatistics === 0) {
+    return (
+      <p>No feedback given</p>
+    )
+  }
+  return (
+    <div>
+      <tr>
+        <td><StatisticLine text='good' /></td>
+        <td><StatisticLine value={good} /></td>
+      </tr>
+      <tr>
+        <td><StatisticLine text='neutral' /></td>
+        <td><StatisticLine value={neutral} /></td>
+      </tr>
+      <tr>
+        <td><StatisticLine text='bad' /></td>
+        <td><StatisticLine value={bad} /></td>
+      </tr>
+      <tr>
+        <td><StatisticLine text='all' /></td>
+        <td><StatisticLine value={good + neutral + bad} /></td>
+      </tr>
+      <tr>
+        <td><StatisticLine text='average' /></td>
+        <td><StatisticLine value={(good - bad)/(good + neutral + bad)} /></td>
+      </tr>
+      <tr>
+        <td><StatisticLine text='positive' /></td>
+        <td><StatisticLine value={good / (good + neutral + bad) *100} /></td>
+      </tr>
+    </div>
+  )
+}
+
+const StatisticLine = ({text, value}) => <div>{text} {value}</div>
+
+const Button = ({ handleClick, text }) => (
+  <button onClick={handleClick}>
+    {text}
+  </button>
+)
+
+const App = () => {
+  const [good, setGood] = useState(0)
+  const [neutral, setNeutral] = useState(0)
+  const [bad, setBad] = useState(0)
+
+  const handleGoodClick = () => {
+    setGood(good + 1)
```

Comment:
```suggestion
    setGood((g) => g+ 1)
```

This ones a bit complicated and better to explain in person, but basically the value of good (the one from this `const [good, setGood] = useState(0)`) only updates once per re-render (not always immediately after a state change). What that means is if you say had your code as 

```js
// lets say good currently is 0
const handleGoodClick = () => {
    setGood(good + 1);
    setGood(good + 1);
}
```

given good is 0, you would now expect good to be 2 right. But good is only updated on rerender, so what you are doing is setGood(`0` + 1) twice. To fix that the setter optionally returns the current value that has been modified since the last render. This is why we do what I did in the example, thinking about life time and state in react is important for not having bugs and keeping thins optimal.